src/api/spec/controllers/webui/subscriptions_controller_spec.rb

```diff
@@ -0,0 +1,38 @@
+require 'rails_helper'
+
+RSpec.describe Webui::SubscriptionsController do
+  describe 'GET #index' do
+    let!(:admin) { create(:admin_user) }
+
+    before do
+      login admin
+      get :index
+    end
+
+    it { expect(response).to have_http_status(:success) }
+    it { expect(response).to render_template(:index) }
+    it { is_expected.to use_before_action(:require_admin) }
+  end
+
+  describe 'PUT #index' do
+    let!(:admin) { create(:admin_user) }
+    let(:params) do
+      {
+        subscriptions: {
+          '0' => { receive: "0", eventtype: "Event::RequestStatechange", receiver_role: "source_maintainer"},
+          '1' => { receive: "0", eventtype: "Event::RequestStatechange", receiver_role: "target_maintainer"},
+          '2' => { receive: "1", eventtype: "Event::RequestStatechange", receiver_role: "creator"},
+          '3' => { receive: "1", eventtype: "Event::RequestStatechange", receiver_role: "reviewer"}
+        }
+      }
+    end
+
+    before do
+      login admin
+      put :update, params: params
+    end
+
+    it { expect(response).to redirect_to(action: :index) }
+    it { is_expected.to use_before_action(:require_admin) }
```

Comment:
@hennevogel Ah I just saw that this comment was in response to what bjoern said.. I've updated this spec so it tests that the data was processed correctly.